nautobot/extras/views.py

```diff
@@ -1487,6 +1491,48 @@ class JobResultView(generic.ObjectView):
     queryset = JobResult.objects.prefetch_related("job_model", "obj_type", "user")
     template_name = "extras/jobresult.html"
 
+    def queryset_to_csv(self, instance):
+        """Format queryset to csv."""
+        csv_data = []
+        headers = JobLogEntry.csv_headers.copy()
+        csv_data.append(",".join(headers))
+
+        for log_entry in instance.logs.all():
+            data = log_entry.to_csv()
+            csv_data.append(csv_format(data))
+
+        return "\n".join(csv_data)
+
+    def get(self, request, *args, **kwargs):
+        """
+        Generic GET handler for accessing an object by PK or slug
+        """
+        instance = get_object_or_404(self.queryset, **kwargs)
+
+        changelog_url = None
+
+        if isinstance(instance, ChangeLoggedModel):
+            changelog_url = instance.get_changelog_url()
+
+        if "export" in request.GET:
```

Comment:
Could probably move this block up a few lines since in the export case we don't need or use `changelog_url`?

Reply:
Good point. Done!